### PR TITLE
feat(FastLED): CFastLED::load + loadChannels + loadScreenMap + FledDispatcher (#3311 PR5)

### PR DESCRIPTION
**Pull request by zackees**
## Summary

**PR5 of #3311.** Adds the bundle-load surface on `CFastLED`, with each method in its own TU so the linker tree-shakes them out of sketches that never reference Fled bundles.

### New surface on CFastLED

- `void load(const Fled& fled)` — composes channels + screenmap + script dispatch
- `void loadChannels(const Fled& fled)` — iterates `fled.channels()` and calls `add()` per
- `void loadScreenMap(const Fled& fled)` — **shipped as a stub** (`FL_WARN_F`); the controller-binding policy decision is deferred to a follow-up. The #3311 spec says \"broadcasts via `EngineEvents::onCanvasUiSet`\", but `onCanvasUiSet` needs a `CLEDController*` and `CFastLED::loadScreenMap` has no specific controller to bind to. Filed for follow-up.

### New singleton

`fl::detail::FledDispatcher` in `src/fl/fled/fled_dispatch.h` — holds the default-empty `wasmLoader` / `microPyLoader` `fl::function` slots. `fledDispatcher()` is a construct-on-first-use accessor — the singleton tree-shakes when no one references it (i.e. `enableWasm` / `enableMicroPy` are never called).

### Tree-shake plumbing

Follows the existing `fl.system.sd` pattern:
- `src/fl/fled/load/`, `load_channels/`, `load_screen_map/` — one TU each
- `src/fl/build/fl.fled.load+.cpp` / `fl.fled.load_channels+.cpp` / `fl.fled.load_screen_map+.cpp` — recursive unity entry points
- `ci/lint_cpp_rs/src/checkers/unity_build.rs` `UNITY_EXPECTED_BUILD_FILES` allowlist extended for the three new build files

Forward decl of `fl::Fled` moved to `src/fl/fled/fwd.h` so `FastLED.h` does not have to include `fled.h`.

## Test plan

- [x] `bash test fled --debug` — green (1/1 DLL, 6 new cases)
- [x] `bash lint --cpp` — green
- [ ] CI matrix

Refs #3311

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added three new API methods to load LED configurations from `.fled` bundles: `load()`, `loadChannels()`, and `loadScreenMap()`.
  * Methods gracefully handle invalid or empty bundles without crashing.

* **Tests**
  * Added comprehensive test coverage for the new bundle loading functionality, including edge cases and null input handling.

### CODE REVIEW

File: src/fl/fled/load/fastled_load.cpp.hpp [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        }
+    } else if (script->kind == fl::ScriptKind::MicroPy) {
+        if (d.microPyLoader) {
+            d.microPyLoader(fled);
+        } else {
+            FL_ERROR_F("MicroPython script detected in .fled bundle but the "
+                     "VM is not linked. Call FastLED.enableMicroPy() in "
+                     "setup() to enable it.");
+        }
+    }
```
[the remark this answers is not in the document]
Comment (zackees) [reply src/fl/fled/load/fastled_load.cpp.hpp]: Moot - the script dispatch block this comment referenced was just removed (see force-push c3ccacc2c). Per follow-up direction wasm/micropy script-host runtimes are design-only for v1, not shipped; load() is now just channels + screenmap.
